packages/auto-spies-core/src/create-auto-spy-from-class.ts

```diff
@@ -85,16 +85,19 @@ function getAllMethodNames(obj: any): string[] {
     const parentObj = Object.getPrototypeOf(obj);
     // we don't want to spy on Function.prototype methods
     if (parentObj) {
-      methods = methods.concat(Object.getOwnPropertyNames(obj));
+      methods = methods.concat(extractMethodsFromObject(obj));
     }
     obj = parentObj;
   }
-
-  const constructorIndex = methods.indexOf('constructor');
-
-  /* istanbul ignore else */
-  if (constructorIndex >= 0) {
-    methods.splice(constructorIndex, 1);
-  }
   return methods;
 }
+
+function extractMethodsFromObject(obj: any) {
+  const propertyDescriptors = Object.getOwnPropertyDescriptors(obj);
+  return Object.entries(propertyDescriptors).reduce((names, [name, descriptor]) => {
```

Comment:
In terms of browser compatibility I don't know if I should use the old Object.keys and the old for loop

Reply:
I believe most of the testers will use either latest chrome or JSDOM.. so... I'm not sure if that should be a concern.

But.. you can always use `keys` if it's not too of a hassle